electroncash/lns.py

```diff
@@ -0,0 +1,607 @@
+            elif isinstance(thing, Exception) and isinstance(exc, list):
+                self.wallet.show_error(str(thing))
+                exc.append(thing)
+            done.set()
+        self.verify_name_asynch(name=lns_string, success_cb=done_cb, error_cb=done_cb, timeout=timeout)
+        if not done.wait(timeout=timeout) or not pb:
+            return
+        return pb;
+
+    def get_lns_names(self, domain=None, inv=False) -> List[Info]:
+        ''' Returns a list of Info objects for verified LNS Names in domain.
+        Domain must be an iterable of addresses (either wallet or external).
+        If domain is None, every verified cash account we know about is returned.
+
+        If inv is True, then domain specifies addresses NOT to include
+        in the results (i.e. eevery verified LNS Name we know about not in
+        domain be returned). '''
+        if domain is None:
+            domain = self.v_by_addr if not inv else set()
+        ret = []
+        seen = set()
+        with self.lock:
+            if inv:
+                domain = set(self.v_by_addr) - set(domain)
+            for addr in domain:
+                infos = self.v_by_addr.get(addr, set())
+                for info in infos:
+                    if info and info not in seen:
+                        seen.add(info)
+                        ret.append(info)
+
+        return ret
+
+    def get_wallet_lns_names(self) -> List[Info]:
+        ''' Convenience method, returns all the verified cash accounts we
+        know about for wallet addresses only. '''
+        return self.get_lns_names(domain=self.wallet.get_addresses())
+
+    def get_external_lns_names(self) -> List[Info]:
+        ''' Convenience method, retruns all the verified cash accounts we
+        know about that are not for wallet addresses. '''
+        return self.get_lns_names(domain=self.wallet.get_addresses(), inv=True)
+
+    def load(self):
+        ''' Note: loading should happen before threads are started, so no lock
+        is needed.'''
+        self._init_data()
+
+        # we do good just by verifying names in contact list upon launch
+        # we do not to persist all names fetched, but I leave the code here for now
+
+        # try:
+        #     dd = self.wallet.storage.get('lns_data', {})
+
+        #     for cashaddr, infos in dd.get('v_by_addr', {}).items():
+        #         self.v_by_addr[Address.from_cashaddr_string(cashaddr)] = set([Info.from_dict(dict) for dict in infos])
+        #     for name, infos in dd.get('v_by_name', {}).items():
+        #         self.v_by_name[name] = set([Info.from_dict(dict) for dict in infos])
+        # except:
+        #     pass
+
+    def save(self, write=False):
+        '''
+        FYI, current data model is:
+
+        self.v_by_addr = defaultdict(set) # dict of addr -> set of txid
+        self.v_by_name = defaultdict(set) # dict of lowercased name -> set of txid
+        '''
+        # ignore saving data for now
```

Comment:
is this needed at all then? imho comment about why we arent saving and delete of code + link to (cashaccount?) side if helpful in future would be better.

Reply:
removed